internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java

```diff
@@ -158,8 +159,18 @@ private static List<Schema.Field> mergeFieldSchemas(Schema existingSchema, Schem
       if (fieldInExistingSchema != null) {
         fieldBuilder.setSchema(generateSupersetSchema(fieldInExistingSchema.schema(), fieldInNewSchema.schema()))
             .setDoc(fieldInNewSchema.doc() != null ? fieldInNewSchema.doc() : fieldInExistingSchema.doc());
+        if (!fieldInNewSchema.hasDefaultValue() && fieldInExistingSchema.hasDefaultValue()) {
```

Comment:
out of curiosity: I wonder whether this would agaisnt the purpose of the method since, in the doc, we mention that we prioritize the new schema over the old schema so we should respect the no default ? (You should update the doc for this new functionality if so I guess)

Reply:
Thank you for the review! I will think more about this and add more details in the doc in the next iteration.

Reply:
Even looking at existing code, it doesn't seems to be always the case that new field overwrites the old. More like a union of the two and when there is any conflict, the new overwrites the old. Agree that we should update the javadoc to make it clear.

e.g. 
`setDoc(fieldInNewSchema.doc() != null ? fieldInNewSchema.doc() : fieldInExistingSchema.doc());`

Reply:
Yeah, updated the doc to reflect the change.